### PR TITLE
feat(issue-platform): add a new post process forwarder for issue platform events

### PR DESCRIPTION
**Pull request by barkbarkimashark**
Adds a new post process forwarder to handle issue occurrence events going through the event stream.

Tested this locally by calling our `issue-occurrence/$` dummy endpoint to produce an issue occurrence event from `occurrence-ingest-consumer` -> `eventstream` -> `post_process_group`. 

It seems testing locally doesn't really go through post-process-forwarder (see https://github.com/getsentry/sentry/blob/404ab7b70ed70e3da27343197e44d6f42ceccfb5/src/sentry/eventstream/snuba.py#L452-L475). But I added a test to verify passing the correct args to the `KafkaEventstream` does properly call `consumer_strategy.dispatch_post_process_group_task`.

### CODE REVIEW

File: src/sentry/runner/commands/devserver.py
Comment (barkbarkimashark) [inline src/sentry/runner/commands/devserver.py:60]: Hmm, do we need to define the `topic` here?
https://github.com/getsentry/sentry/blob/41b8af967e17f55796569f299794d9e67cf59b12/src/sentry/runner/commands/run.py#L327-L331

File: src/sentry/runner/commands/devserver.py
Comment (lynnagara) [inline src/sentry/runner/commands/devserver.py:55]: We also have `generic-metrics` so I think it'll be good to call this something more specific for example:
`post-process-forwarder-generic-events`, `post-process-forwarder-issue-platform`, `post-process-forwarder-search-issues`
(unrelated but i've seen variants of these names floating around and i'm not sure which is the canonical name - can things be standardised on one?)

File: src/sentry/runner/commands/devserver.py
Comment (rjo100) [reply src/sentry/runner/commands/devserver.py:60]: Should be handled internally if we define `entity` correctly
https://github.com/getsentry/sentry/blob/master/src/sentry/eventstream/kafka/backend.py#L296

Comment (lynnagara) [reply src/sentry/runner/commands/devserver.py:60]: It is a bit unclear I know... but it gets picked up from https://github.com/getsentry/sentry/blob/a813a868b7bc423785f082758c4fd5eecee96386/src/sentry/eventstream/kafka/backend.py#L295-L296 and the others don't define is so I'd say it's fine to omit

**Review by wedamija — APPROVED** [review]

**Review by lynnagara — CHANGES_REQUESTED** [review]
This seems like it can break deverserver since Snuba is not yet bootstrapping and producing the commit log topic. Please either
1) remove the devserver changes from this PR and add them later, or
2) create + merge the Snuba change first and mark that as a dependency here

File: src/sentry/runner/commands/devserver.py
Comment (barkbarkimashark) [reply src/sentry/runner/commands/devserver.py:55]: Yup, good callout. Was planning to wait till we sync with OPS before merging this in to avoid any incidents.

Comment (barkbarkimashark) [reply src/sentry/runner/commands/devserver.py:55]: Hmm, was gonna create a new PR to revert the revert that added the commit log to prod settings, but that looks like it hasn't merged https://github.com/getsentry/sentry/pull/45014

Not sure what's the status of prod snuba but won't be merging this until snuba is in a known state.

Comment (barkbarkimashark) [reply src/sentry/runner/commands/devserver.py:55]: @lynnagara I think we should be good to deploy this based on this comment from Ben
https://github.com/getsentry/ops/pull/6180#issuecomment-1442180107

Comment (barkbarkimashark) [reply src/sentry/runner/commands/devserver.py:55]: Or maybe you're referring to this PR which should start producing to the topic https://github.com/getsentry/snuba/pull/3775/files.

Or maybe we should merge this first which consumes from the commit log topic before producing?

Comment (lynnagara) [reply src/sentry/runner/commands/devserver.py:55]: Yep. https://github.com/getsentry/snuba/pull/3775](https://github.com/getsentry/snuba/pull/3775/

We should not merge this first or it may break devserver

**Comment by lynnagara** [comment]
Were you able to run this end-to-end and verify that issue platform post processing is being called?

**Review by lynnagara — APPROVED** [review]

**Pull request merged**
